test/authn/Authn.test.ts

```diff
@@ -0,0 +1,59 @@
+import { keccak256 } from 'js-sha3'
+import { PrivateKey, PrivateKeyBundle, Signature } from '../../src/crypto'
+import Authenticator from '../../src/authn/Authenticator'
+import Token from '../../src/authn/Token'
+import { hexToBytes } from '../../src/crypto/utils'
+import { newWallet } from '../helpers'
+import { Wallet } from 'ethers'
+
+describe('authn', () => {
+  let authenticator: Authenticator
+  let privateKey: PrivateKey
+  let wallet: Wallet
+
+  beforeEach(async () => {
+    wallet = newWallet()
+    const bundle = await PrivateKeyBundle.generate(wallet)
+    privateKey = bundle.identityKey
+    authenticator = new Authenticator(privateKey)
+  })
+
+  it('can create a token', async () => {
+    const timestamp = new Date()
+    const token = await authenticator.createToken(timestamp)
+
+    expect(token.authData.walletAddr).toEqual(
+      privateKey.publicKey.getEthereumAddress()
+    )
+    expect(token.authData.createdNs).toEqual(+timestamp * 1000)
+    expect(token.identityKey.timestamp).toEqual(privateKey.publicKey.timestamp)
+    expect(token.identityKey.signature).toEqual(privateKey.publicKey.signature)
+    expect(token.identityKey.secp256k1Uncompressed).toEqual(
+      privateKey.publicKey.secp256k1Uncompressed
+    )
+  })
+
+  it('rejects unsigned identity keys', async () => {
```

Comment:
Interesting, you already have a test for the key signature, not sure how that works.